src/pubkey.cpp

```diff
@@ -68,8 +68,8 @@ bool CPubKey::Decompress() {
         return false;
 #ifdef USE_SECP256K1
     int clen = size();
-    int ret = secp256k1_ecdsa_pubkey_decompress((unsigned char*)begin(), &clen);
-    assert(ret);
+    if (secp256k1_ec_pubkey_decompress((unsigned char*)begin(), &clen) == 0)
+        return false;
```

Comment:
This code (everything in #ifdef USE_SECP256k1) is currently unused, and should just be removed until it's actually supported.